File: lib/features/drive/data/drive_upload_payload.dart
import 'dart:typed_data';

class PickedDriveFile {
  const PickedDriveFile({
    required this.name,
    required this.contentType,
    required this.sizeBytes,
    required this.bytes,
  });

  final String name;
  final String contentType;
  final int sizeBytes;
  final Uint8List bytes;

  bool get hasSupportedExtension {
    final lower = name.toLowerCase();
    return lower.endsWith('.pdf') ||
        lower.endsWith('.ppt') ||
        lower.endsWith('.pptx') ||
        lower.endsWith('.doc') ||
        lower.endsWith('.docx') ||
        lower.endsWith('.zip');
  }

  bool get hasReadableContent => sizeBytes > 0 && bytes.isNotEmpty;
}

File: SourceBaseBackend/Sources/SourceBaseBackend/Drive/DriveUploadPayload.swift
import Foundation

public struct PickedDriveFile: Sendable {
    public let name: String
    public let contentType: String
    public let sizeBytes: Int
    public let data: Data?
    public let fileURL: URL?

    public init(name: String, contentType: String, sizeBytes: Int, data: Data) {
        self.name = name
        self.contentType = contentType
        self.sizeBytes = sizeBytes
        self.data = data
        self.fileURL = nil
    }

    public init(name: String, contentType: String, sizeBytes: Int, fileURL: URL) {
        self.name = name
        self.contentType = contentType
        self.sizeBytes = sizeBytes
        self.data = nil
        self.fileURL = fileURL
    }

    public var hasSupportedExtension: Bool {
        DriveUploadService.isSupportedFileName(name)
    }

    public var hasReadableContent: Bool {
        sizeBytes > 0 && (data?.isEmpty == false || fileURL != nil)
    }
}
